src/misc/utils.ts

```diff
@@ -29,13 +29,13 @@ export function assert(
             let part: string;
 
             if (detail instanceof ASTNode) {
-                part = detail.type + " #" + detail.id;
+                part = pp(detail);
 
                 nodes.push(detail);
-            } else if (detail instanceof TypeNode) {
-                part = detail.pp();
+            } else if (detail instanceof Array) {
```

Comment:
This case can be removed after we add the additional ifs in pp().

Reply:
Addressed in https://github.com/ConsenSys/solc-typed-ast/commit/6e612113efada422eb324ed77841541855af8c10